File: 백준/Silver/16435. 스네이크버드/스네이크버드.py
# 과일 하나 먹으면 길이 1 늘어남
# i번째 과일 높이 h
# 자신의 길이보다 작거나 같은 높이 => 먹을 수 있음

N, L = map(int,input().split())

fruits = list(map(int,input().split()))
fruits.sort()
for f in fruits :
    if f <= L :
        L += 1
print(L)

File: 백준/Silver/16435. 스네이크버드/스네이크버드.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int N = scanner.nextInt();
        int L = scanner.nextInt();

        int[] fruits = new int[N];
        for (int i = 0; i < N; i++) {
            fruits[i] = scanner.nextInt();
        }

        Arrays.sort(fruits);

        for (int f : fruits) {
            if (f <= L) {
                L++;
            }
        }

        System.out.println(L);
    }
}